lib/create-identity-response-handler.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * The contents of this file are only relevant if your service connects
+ * to GOV.UK Verify using the Verify Service Provider (VSP) without a Matching
+ * Service Adapter (MSA).
+ *
+ *
+ * create-identity-response-handler
+ *
+ * Helper function to make handling callbacks from
+ * `passport.authenticate('verify', callback)` easier.
+ *
+ * This is necessary because there are a number of different scenarios
+ * that services need to handle when dealing with Verify responses.
+ * passport.js isn't aware of these, so the interface it provides is not ideal.
+ */
+/** */
+
+import { TranslatedIdentityResponseBody, Scenario } from './verify-service-provider-api/translated-response-body'
+
+/**
+ * The callbacks in the `IdentityResponseScenarios` interface define scenarios that
+ * can occur when handling a response from GOV.UK Verify.
+ *
+ * Services should handle each of these scenarios appropriately.
+ */
+export interface IdentityResponseScenarios {
+
+  /**
+   * Called when handling an `IDENTITY_VERIFIED` response. Your callback should
+   * redirect the user to the appropriate page so they can begin using your service.
+   *
+   * The callback will be passed an `identity` object, which will be the output from
+   * your `handleIdentity` callback.
+   */
+  onIdentityVerified: (user: any) => any,
+
+  /**
+   * Called when the user failed to authenticate with Verify,
```

Comment:
```suggestion
   * Called when the user failed to authenticate with the identity provider,
```
The user authenticates by logging into their account with the identity provider.

Reply:
If they've failed to authenticate with an IDP, then they've also failed to authenticate with Verify.  An RP developer doesn't really need to understand that distinction to work on integrating with passport-verify; I think we'll just be muddying the water if we give too much detail.

Reply:
```suggestion
   * Called when the user failed to authenticate,
```
I agree on muddying the water with too much detail, so then let's leave the details out rather than putting in incorrect information. It's incorrect to say users authenticate with Verify and we shouldn't encourage the phrasing.

Reply:
I stand corrected; fair enough then!  'Failed to authenticate' is clear.